### PR TITLE
Use plugin_lang_get to display repo settings keys, trans bool values

### PR DESCRIPTION
**Pull request by obmsch**
Instead of just displaying the raw keys, be user friendly and show the lang
strings as repo_update does. For bool values use trans_bool to get a nice
check mark.

### CODE REVIEW

File: Source/pages/repo_manage_page.php [hunk 1]
```diff
@@ -121,8 +122,12 @@ function convert_to_key_value( $p_array ) {
 			foreach( $t_formatted_array as $t_key => $t_value ) {
 		?>
 		<tr>
-			<td class="category"><?php echo $t_key ?></td>
+			<td class="category"><?php echo plugin_lang_get( $t_key, $t_vcs->basename ) ?></td>
+<?php if( is_bool( $t_value ) ) { ?>
+			<td><?php echo trans_bool( $t_value ) ?></td>
+<?php } else { ?>
 			<td><?php echo $t_value ?></td>
+<?php } ?>
```
Comment (dregad) [inline Source/pages/repo_manage_page.php]: To make the code easier to read, I would suggest to use a ternary operator here, i.e.
``` php
<td><?php echo is_bool( $t_value ) ? trans_bool( $t_value ) : $t_value ?></td>
```

**Comment by obmsch** [comment]
Rebased, and hopefully adressed all your comments.
Local function "plugin_lang_get_defaulted" eventually moves to mantisbt core.
A lot of implications and dependencies, surely beyond the scope of this PR.

**Comment by dregad** [comment]
Hello Martin, and thanks for the revised PR.

> hopefully adressed all your comments

Actually you missed point 2 in https://github.com/mantisbt-plugins/source-integration/pull/215#pullrequestreview-43306872, i.e. splitting the language strings between _label_ and _help text_.

> Local function "plugin_lang_get_defaulted" eventually moves to mantisbt core.
> A lot of implications and dependencies, surely beyond the scope of this PR.

That makes perfect sense, I'll prepare a MantisBT PR so it becomes part of core in a future version. 

In the meanwhile, to keep the code clean I think it would be better to move this function into _Source.API.php_, and also wrap it around an _if function_exists_ block to facilitate the transition.

**Comment by obmsch** [comment]
Hello Damien,
moving (and wrapping) "plugin_lang_get_defaulted" to Source.API.php makes sense, will look into that later.
I am not sure what you have in mind on the string splitting issue. Are you asking for something like:
$s_plugin_SourceGithub_master_branch = 'Primary Branches<br/><span class="small">(comma-separated list or "*" for all branches)</span>';
gets

a) $s_plugin_SourceGithub_master_branch = 'Primary Branches';
   $s_plugin_SourceGithub_master_branch_help = '<br/><span class="small">(comma-separated list or "*" for all branches)</span>';

b) $s_plugin_SourceGithub_master_branch_label = 'Primary Branches';
   $s_plugin_SourceGithub_master_branch_help = '<br/><span class="small">(comma-separated list or "*" for all branches)</span>';

c) $s_plugin_SourceGithub_master_branch_label = 'Primary Branches';
   $s_plugin_SourceGithub_master_branch = 'Primary Branches<br/><span class="small">(comma-separated list or "*" for all branches)</span>';

As repo_manage_page and repo_update_page use the same keys to lookup the lang strings

a) Would require an additional lookup at least in all "$t_vcs->update_repo_form( $t_repo )" variants, for a key(help) which  may (intentionally) not exist.
b) In principle the same as a), but the first lookup has to be adjusted too.
c) Despite more effort on the lang maintenance, translations, this would be the easiest way to go for. No side effects, only  repo_manage_page affected.
   instead of
 	# Extract the descriptive part of the $t_key lang entry
	list( $t_description ) = explode( "<br", plugin_lang_get_defaulted( $t_key, $t_vcs->basename, $t_key ) );
        ... echo $t_description ...
   then simply
	... echo plugin_lang_get_defaulted( $t_key . "_label", $t_vcs->basename, $t_key ) ...

Thoughts?

**Comment by dregad** [comment]
From a design perspective, I never liked these labels including help text with HTML formatting, so I thought that this would be a good opportunity to get rid of them.

As for implementation, I was actually thinking along the lines of your option a) which I think is cleaner than c); b) does not make sense IMO. 

I must admit that this was just an idea, I did not fully consider or analyze the implications on the code base and development effort. That being said, I'm not overly concerned about the extra plugin_lang_get() calls, as this would be a one time change.

Anyway this is diverging from your PR's original scope, so maybe it would be better to track this refactoring separately.

**Comment by obmsch** [comment]
One more update to the PR. Made "basename" otional and last param in "plugin_lang_get_defaulted".
More convenient, I guess only a few cases (like repo_manage_page) where current basename differs
from the target basename.

For now help text are shown too (minor issue) and will be tackled, when issue #228 gets fixed.

**Comment by dregad** [comment]
> One more update to the PR. Made "basename" otional and last param in "plugin_lang_get_defaulted".

Indeed, much better this way, and closer to *lang_get_defaulted()*'s signature too.

**Comment by dregad** [comment]
> I'll prepare a MantisBT PR so it becomes part of core in a future version.

https://mantisbt.org/bugs/view.php?id=26747

**Pull request merged**